Synapse/Api/Map.cs

```diff
@@ -105,6 +106,43 @@ public void AnnounceScpDeath(string scp)
             GlitchedCassie(text);
         }
 
+        public void AnnounceScpDeath(string scp, ScpReconfType deathType, string Unit = "UNKNOWN")
+        {
+            string cause;
+            switch (deathType)
+            {
+                case ScpReconfType.Tesla:
+                    cause = $". SCP {scp} SUCCESSFULLY TERMINATED BY AUTOMATIC SECURITY SYSTEM";
+                    break;
+                case ScpReconfType.Nuke:
+                    cause = $". SCP {scp} SUCCESSFULLY TERMINATED BY ALPHA WARHEAD";
+                    break;
+                case ScpReconfType.Decontamination:
+                    cause = $". SCP {scp} LOST IN DECONTAMINATION SEQUENCE";
+                    break;
+                case ScpReconfType.ByMtf:
+                    cause = $". SCP {scp} SUCCESSFULLY TERMINATED . CONTAINEDSUCCESSFULLY CONTAINMENTUNIT {Unit}";
+                    break;
+                case ScpReconfType.ByChi:
+                    cause = $". SCP {scp} SUCCESSFULLY TERMINATED . BY CHAOSINSURGENCY";
+                    break;
+                case ScpReconfType.ByRsc:
+                    cause = $". SCP {scp} SUCCESSFULLY TERMINATED . BY SCIENCE PERSONNEL";
+                    break;
+                case ScpReconfType.ByCdp:
+                    cause = $". SCP {scp} SUCCESSFULLY TERMINATED . BY CLASSD PERSONNEL";
+                    break;
+                case ScpReconfType.ByUnknown:
+                    cause = $". SCP {scp} SUCCESSFULLY TERMINATED . CONTAINMENTUNIT UNKNOWN";
+                    break;
+                case ScpReconfType.Unspecified:
+                default:
+                    cause = $". SCP {scp} SUCCESSFULLY TERMINATED . TERMINATION CAUSE UNSPECIFIED";
+                    break;
+            }
```

Comment:
You may want to consider using a switch-assignment, see https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-8#switch-expressions